src/StreamJsonRpc.Tests/JsonRpcTests.cs

```diff
@@ -993,6 +995,17 @@ public async Task InvokeWithSingleObjectParameter_SendingExpectedObjectAndCancel
         Assert.Equal(7, sum);
     }
 
+    [Fact]
+    public async Task AddLocalRpcTarget_UseSingleObjectParameterDeserialization()
+    {
+        var streams = FullDuplexStream.CreatePair();
+        var rpc = new JsonRpc(streams.Item1, streams.Item2);
+        rpc.AddLocalRpcTarget(new Server(), new JsonRpcTargetOptions { UseSingleObjectParameterDeserialization = true });
+        rpc.StartListening();
+
+        Assert.Equal(3, await rpc.InvokeAsync<int>(nameof(IServer.InstanceMethodWithSingleObjectParameterButNoAttribute), new XAndYFields { x = 1, y = 2 }));
```

Comment:
Calling `InvokeAsync` isn't using named arguments at all, so I don't think it's actually exercising your change. Does this test fail without your product changes (or when you don't set your new property to true on Line 1003)? I'm guessing it would pass.

You should call `InvokeWithParameterObjectAsync` here.

Reply:
fixed